### PR TITLE
`General`: Example User Management SQL Scripts And Liquibase Update

### PR DESCRIPTION
**Pull request by marc-fett**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://confluence.aet.cit.tum.de/spaces/AP/pages/257786006/Language+Guidelines+Client).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311714/PR+Guidelines).

#### Server

- [x] **Important**: I implemented the changes with a [very good performance](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311718/Performance+Guidelines) and prevented too many (unnecessary) and too complex database calls.
- [x] I **strictly** followed the principle of **data economy** for all database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311877/Server+Guidelines).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context

This PR redefines the initial database schema for user management using Liquibase changelogs.

It ensures clean setup of:
- users
- research groups
- user roles

It introduces a clean, modular user management schema using UUID-based primary keys.
It supports flexible role assignment, including global and research group-specific roles.

In addition, this PR introduces a reusable import script for inserting structured example data to support development and testing.

### Description

This PR includes the following structural and test data changes:

#### Liquibase schema updates:
Modular Liquibase change sets for:
- `users` with flexible attributes and auditing timestamps
- `research_groups` with extended metadata
- `user_research_group_roles` with UUID primary key, foreign keys, and enum-based `role`
- `NOT NULL` constraints and `TEXT` types for long user input fields
- foreign key constraints and unique role assignments per (user, research group, role)

#### Example test data:
- `01_users.sql`: Inserts 5 example users (admin, professor, TUM applicant, external applicant, reviewer) with various edge cases and realistic values.
- `02_research_groups.sql`: Inserts 2 example research groups and assigns selected users to them.
- `03_user_roles.sql`: Assigns user roles with optional research group context (e.g., global ADMIN, group-specific PROFESSOR).
- `import-testdata.sh`: Bash script for importing the test data using the local MySQL setup. Automatically runs all `.sql` files from the same directory.

### Steps for Testing

**Prerequisites:**

All the commands in the README.md under "Development" > "Setting up the Database & Importing Test Data".
1. Ensure the local MySQL instance (from Docker Compose) is running.
2. Confirm the `tumapply` database exists.
3. Run the script:  
   ```bash
   ./src/main/resources/testdata/import-testdata.sh
4. Verify that global roles (e.g., ADMIN, APPLICANT) and scoped roles (e.g., PROFESSOR) are imported and assigned correctly.

### Review Progress

<!-- Each PR should be reviewed by at least one other developers. The code, the functionality (= manual test). -->
<!-- The reviewer or author check the following boxes depending on what was reviewed or tested. All boxes should be checked before merge. -->
<!-- You can add additional checkboxes if it makes sense to only review parts of the code or functionality. -->
<!-- When changes are pushed, uncheck the affected boxes. (Not all changes require full re-reviews.) -->

#### Code Review

- [x] Code Review 1

#### Manual Tests

- [x] Liquibase migrations run without errors
- [x] Created tables match defined structure
- [x] Data from test files appears correctly in the database

### CODE REVIEW

**Review by MoSchmidt — APPROVED** [review]
I tested to load the test-data in the databse and it worked.

Note: to achieve this on Windows, I would suggest to use WSL and install the mySQL client there

**Review by MoSchmidt — CHANGES_REQUESTED** [review]
Database schema does not match the prefeined structure. E.g. users should not conatin any files. Those should be stored in a seperate applicants table

File: gradle/liquibase.gradle [hunk 1]
```diff
@@ -19,22 +19,22 @@ if (!project.hasProperty("runList")) {
...
 liquibase {
   activities {
-    register("main") {
-      driver = "com.mysql.cj.jdbc.Driver"
-      url = "jdbc:mysql://localhost:3306/TUMApply"
-      username = "root"
-      changelogFile = "src/main/resources/config/liquibase/master.xml"
-      logLevel = "debug"
-      classpath = "src/main/resources/"
+    main {
```
Comment (ani-ib) [inline gradle/liquibase.gradle:22]: I noticed the change from register("main") to main { ... }. Is there a specific reason this was changed?

**Comment by marc-fett** [comment]
> The database schema does not match the predefined structure. E.g., users should not contain any files. Those should be stored in a separate applicants table.

I would change the entity "User.java" instead of the initial schema since we need the cv, bachelor certificate, ... ids in the user right? Otherwise we can't have default documents of a user. @MoSchmidt

File: gradle/liquibase.gradle (hunk 1 above)
Comment (marc-fett) [reply gradle/liquibase.gradle:22]: I changed it to the syntax of the official docu: https://contribute.liquibase.com/extensions-integrations/directory/integration-docs/gradle/

**Comment by ani-ib** [comment]
@marc-fett We intentionally separated the applicant-specific fields into a dedicated applicants table to avoid populating the users table with numerous columns that would remain unused (i.e., null) for professors and other user types. This structure ensures cleaner data separation and better reflects the different requirements of each user role. Stephan also suggested following this approach to keep the data model clean and maintainable.

**Comment by marc-fett** [comment]
> @marc-fett We intentionally separated the applicant-specific fields into a dedicated applicants table to avoid populating the users table with numerous columns that would remain unused (i.e., null) for professors and other user types. This structure ensures cleaner data separation and better reflects the different requirements of each user role. Stephan also suggested following this approach to keep the data model clean and maintainable.

@ani-ib yes, I removed some fields in the newest commit to avoid redundancy and unnecessary null values.

**Review by ani-ib — CHANGES_REQUESTED** [review]
Left a few important comments

File: gradle/liquibase.gradle [hunk 2]
```diff
@@ -19,22 +19,22 @@ if (!project.hasProperty("runList")) {
...
+      driver "com.mysql.cj.jdbc.Driver"
+      url "jdbc:mysql://localhost:3306/tumapply"
+      username "root"
+      changelogFile "src/main/resources/config/liquibase/master.xml"
+      logLevel "debug"
+      classpath "src/main/resources/"
     }
-    register("diffLog") {
```
[the remark this answers is not in the document]
Comment (marc-fett) [reply gradle/liquibase.gradle:30]: I changed it to the syntax of the official docu: https://contribute.liquibase.com/extensions-integrations/directory/integration-docs/gradle/

File: src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml [hunk 3]
```diff
@@ -1,92 +1,132 @@
...
+      <column name="last_name" type="VARCHAR(255)">
+        <constraints nullable="false"/>
+      </column>
+      <column name="gender" type="VARCHAR(255)"/>
+      <column name="nationality" type="VARCHAR(255)"/>
+      <column name="website" type="TEXT"/>
+      <column name="cv_file_id" type="CHAR(36)"/>
+      <column name="bachelor_certificate_id" type="CHAR(36)"/>
+      <column name="master_certificate_id" type="CHAR(36)"/>
+      <column name="study_degree" type="VARCHAR(255)"/>
```
[the remark this answers is not in the document]
Comment (marc-fett) [reply src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml]: Removed some columns in the changelog file and in confluence.

File: src/main/resources/testdata/01_users.sql [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+-- Preconditions:
+--   - No foreign key constraints violated (research_groups will be inserted later)
+--   - avatar_file_id, cv_file_id, bachelor_certificate_id, master_certificate_id may be NULL
+-- =============================================
+
+-- Clean up existing users
+DELETE
+FROM users
+WHERE user_id IN ('00000000-0000-0000-0000-000000000101', '00000000-0000-0000-0000-000000000102',
+                  '00000000-0000-0000-0000-000000000103', '00000000-0000-0000-0000-000000000104');
```
[the remark this answers is not in the document]
Comment (marc-fett) [reply src/main/resources/testdata/01_users.sql]: No, there was no reason, just added it later and forgot to add it to the delete statement, thanks.

File: gradle/liquibase.gradle (hunk 1 above)
Comment (az108) [reply gradle/liquibase.gradle:22]: Can we please keep register here?

File: src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml [hunk 5]
```diff
@@ -1,92 +1,132 @@
...
-        <createTable tableName="jhi_user_authority">
-            <column name="user_id" type="varchar(100)">
-                <constraints nullable="false"/>
-            </column>
-            <column name="authority_name" type="varchar(50)">
-                <constraints nullable="false"/>
-            </column>
-        </createTable>
+  <changeSet id="003_create_user_roles_and_logins" author="marc-fett">
+    <createTable tableName="user_roles">
```
[the remark this answers is not in the document]
Comment (marc-fett) [reply src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml:67]: We will discuss this in the upcoming meeting.

File: src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml [hunk 6]
```diff
@@ -1,92 +1,109 @@
...
-            <column name="name" type="varchar(50)">
-                <constraints primaryKey="true" nullable="false"/>
-            </column>
-        </createTable>
+  <changeSet id="002_create_research_groups" author="marc-fett">
+    <createTable tableName="research_groups">
+      <column name="research_group_id" type="CHAR(36)">
+        <constraints nullable="false" primaryKey="true"/>
+      </column>
+      <column name="head" type="CHAR(36)">
```
Comment (ani-ib) [inline src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml:46]: Would it make more sense to use VARCHAR instead of CHAR here, since we’re storing string values and not actual UUIDs? Using VARCHAR might be a bit more flexible if the length isn’t strictly fixed. What do you think?

**Review by ani-ib — CHANGES_REQUESTED** [review]
Code looks good overall - just suggested some changes based on the updated type of the "head" field in the ResearchGroup Entity

File: src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml (hunk 6 above)
Comment (marc-fett) [reply src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml:46]: Changed to VARCHAR (255), thanks 👍🏼

File: src/main/resources/testdata/02_research_groups.sql [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+                              email,
+                              website,
+                              school,
+                              description,
+                              default_field_of_studies,
+                              street,
+                              postal_code,
+                              city)
+VALUES ('00000000-0000-0000-0000-000000000001',
+        '00000000-0000-0000-0000-000000000102',
```
[the remark this answers is not in the document]
Comment (marc-fett) [reply src/main/resources/testdata/02_research_groups.sql:29]: Yes, changed it to a Professor name in both cases, thanks 👍🏼

**Review by ani-ib — APPROVED** [review]
Code LGTM - I followed the steps in README for testing, and the tables match the defined structure in the database

**Comment by ani-ib** [comment]
Please adjust the PR description as well (it contains outdated changes such as GPA columns)

File: src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml [hunk 8]
```diff
@@ -1,92 +1,109 @@
...
-            <column name="lang_key" type="varchar(10)"/>
-            <column name="created_by" type="varchar(50)">
-                <constraints nullable="false"/>
-            </column>
-            <column name="created_date" type="timestamp"/>
-            <column name="last_modified_by" type="varchar(50)"/>
-            <column name="last_modified_date" type="timestamp"/>
-        </createTable>
+  <changeSet id="001_create_users" author="marc-fett">
+    <createTable tableName="users">
```
Comment (ani-ib) [inline src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml:8]: We still need to add the "birthday" and "selected_language" columns, as discussed yesterday based on the required fields from the AOM model.

**Review by ani-ib — CHANGES_REQUESTED** [review]
I’ve added another comment based on the updated User table schema discussed yesterday.

File: src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml (hunk 8 above)
Comment (marc-fett) [reply src/main/resources/config/liquibase/changelog/00000000000000_initial_schema.xml:8]: Added birthday and selected language in the documentation on Confluence and the code.

**Review by ani-ib — APPROVED** [review]
Code LGTM - I followed the steps in README for testing, and the tables match the defined structure in the database
